File: platforms/ios/foundation/Sources/NativeblocksFoundation/block/Item.swift
import NativeblocksCompiler
import NativeblocksRuntime
import SwiftUI

@Block(
    name: "Item",
    keyType: "nativeblocks/item",
    description: "One item of fixed content inside a list.",
    scope: "LIST",
    version: 1,
    versionName: "1"
)
struct Item<Content: View>: View {
    var blockContext: BlockContext? = nil

    var describeScope: Any = ()

    @BlockSlot(description: "The content of this item.")
    var content: (BlockIndex) -> Content

    var body: some View {
        return content(-1)
    }
}

File: platforms/android/foundation/foundation/src/main/java/io/nativeblocks/foundation/block/Item.kt
package io.nativeblocks.foundation.block

import androidx.compose.runtime.Composable
import io.nativeblocks.compiler.type.Block
import io.nativeblocks.compiler.type.BlockIndex
import io.nativeblocks.compiler.type.BlockSlot
import io.nativeblocks.foundation.util.LazyListDescribeScope
import io.nativeblocks.runtime.api.provider.block.BlockContext
import io.nativeblocks.runtime.api.provider.block.NONE_INDEX

@Block(
    keyType = "nativeblocks/item",
    name = "Item",
    description = "One item of fixed content inside a list.",
    scope = "LIST",
    version = 1,
    versionName = "1",
)
internal fun item(
    blockContext: BlockContext,
    describeScope: Any,
    @BlockSlot(
        description = "The content of this item.",
    ) content: @Composable (index: BlockIndex, scope: Any) -> Unit,
) {
    val listScope = describeScope as? LazyListDescribeScope ?: return
    listScope.item(blockContext.block.key) { scope -> content(NONE_INDEX, scope) }
}
